### DISCUSSION TITLE
Maximum idle and persistence

### DISCUSSION THREAD

**Discussion by schall1337** [infinispan/infinispan#13037] [Q&A] [answered]
Hello everyone,
I have a question regarding maximum idle and persistence. 

From the docs (https://infinispan.org/docs/15.0.x/titles/configuring/configuring.html#eviction-and-expiration_configuring-memory-usage) I can see that "Maximum idle expiration does not currently support caches with persistent storage".

My cache configuration looks like that with persistence to file store. In my Quarkus application I am writing entries to the cache with maximum idle (120 seconds). After the idle time they seem to be removed correctly from the cache even though persistence is configured.

Question: Why do the docs specify that it is not supported even though it seems to work? 

`{
	"replicated-cache": {
		"mode": "SYNC",
		"statistics": true,
		"encoding": {
			"media-type": "application/x-protostream"
		},
		"memory": {
			"storage": "OFF_HEAP"
		},
		"persistence": {
			"passivation": false,
			"file-store": {
				"data": {
					"path": "data"
				},
				"index": {
					"path": "index"
				}
			}
		}
	}
}`

**Comment by wburns** [comment, marked as answer]
While it may work in your case, sadly that is the only case it works (that being it expiring with no access). Max idle is effectively lifespan expiration when it comes to a store. We do not update the freshness timestamps as this would be extremely expensive with all provided store implementations, causing a read to be as expensive as put if max idle was implemented in such a way.
